pyerrors/correlators.py

```diff
@@ -343,6 +344,36 @@ def GEVP(self, t0, ts=None, sort="Eigenvalue", **kwargs):
         else:
             return reordered_vecs
 
+    def error_gevp(self, t0):
+        """Solves the GEVP with statistical errors.
+        Due to numerical instabilities and sign conventions the result does not necessarily agree with the standard GEVP method.
+
+        Parameters
+        ----------
+        t0 : int
+            The time t0 for the right hand side of the GEVP
+
+        The returned value is in the same form as the one of corr.GEVP(), but the entries are now Obs.
+        """
+        output = [[[] for i in range(self.T)] for j in range(self.N)]
+        G0 = np.vectorize(lambda x: x)(self[t0])
+        chol = cholesky(G0)  # This will automatically report if the matrix is not pos-def
+        chol_inv = inv(chol)
+        for ts in range(self.T):
+            try:
+                Gt = np.vectorize(lambda x: x)(self[ts])
+                new_matrix = chol_inv @ Gt @ chol_inv.T
+                for state in range(self.N):
+                    ev = chol_inv.T @ eigh(new_matrix)[1][:, state]
+                    ev = np.array([e / np.sqrt(ev @ G0 @ ev.T) for e in ev])
+                    [e.gamma_method() for e in ev]  # While this is slow, there is a large risk a user will project without calculating the errors
+                    output[state][ts] = ev
+            except Exception:  # The above code can fail because of linalg-errors or because the entry of the corr is None
```

Comment:
I think it would be better to not expect a bare `Exception` but instead specify which type of exception you expect.

Reply:
True, but there are multiple possible exceptions. There could be a linalg error or Gt could be None.

Reply:
Yes, that is more elegant, but i just do not know, what the specific errors might be.

Reply:
I would suspect `TypeError` if you try to multiply an `Obs` matrix with a `None` matrix?
```
TypeError: unsupported operand type(s) for *: 'float' and 'NoneType'
```

Reply:
So it would be `except (LinAlgError, TypeError):` I guess? It is still very broad, but it is better than catching everything.

Reply:
Yes, let's try that. Thanks!